virttest/qemu_vm.py

```diff
@@ -458,28 +458,30 @@ def add_serial(devices, name, filename):
             if (not devices.has_option("chardev") or
                     not any(devices.has_device(dev)
                             for dev in ("isa-serial", "sclpconsole", "spapr-vty"))):
-                return " -serial unix:'%s',server,nowait" % filename
+                cmd = " -serial unix:'%s',server,nowait" % filename
+                return StrDev('SER-%s' % name, cmdline=cmd)
+            char_params = params.object_params(name)
+            backend = char_params.get("chardev_backend", "socket")
```

Comment:
@luckyh Thanks, renamed with `unix`, and modify add_chardev() to generate params `port` and `host` instead of `path` for udp and tcp, https://github.com/avocado-framework/avocado-vt/pull/1587/files#diff-3f2317089f10636f453dca6c4f880f6cR498